@here/harp-mapview/lib/Utils.ts

```diff
@@ -162,32 +179,42 @@ export namespace MapViewUtils {
         const groundDistance = calculateDistanceToGroundFromZoomLevel(mapView, zoomLevel);
         const cameraHeight = groundDistance + (elevation ?? 0);
 
-        if (mapView.geoMaxBounds && worldTarget) {
-            const { distance } = constrainTargetAndDistanceToViewBounds(
-                worldTarget,
+        if (mapView.geoMaxBounds) {
+            // If map view has maximum bounds set, constrain camera target and distance to ensure
+            // they remain within bounds.
+
+            // tslint:disable-next-line: deprecation
+            const oldTarget = getTargetAndDistance(projection, camera, elevationProvider).target;
+            const newCamPos = setWorldPosHeight(
```

Comment:
I have problems understanding all that code. Why do we use the camera height at all and then compute the distance? Can we not directly compute the desired distance and constrain it?

Reply:
changed as discussed.